storage/isolation_manager.rs

```diff
@@ -212,17 +137,75 @@ impl IsolationManager {
     }
 }
 
+fn resolve_concurrent(
+    commit_record: &CommitRecord,
+    predecessor_sequence_number: SequenceNumber,
+    predecessor_window: &TimelineWindow<TIMELINE_WINDOW_SIZE>,
+) -> Result<(), IsolationError> {
+    let commit_dependency = match predecessor_window.get_status(predecessor_sequence_number) {
+        CommitStatus::Empty => unreachable!("A concurrent status should never be empty at commit time"),
+        CommitStatus::Pending(predecessor_record) => match commit_record.compute_dependency(predecessor_record) {
+            CommitDependency::Independent => CommitDependency::Independent,
+            result => {
+                if predecessor_window.await_pending_status_commits(predecessor_sequence_number) {
+                    result
+                } else {
+                    CommitDependency::Independent
+                }
+            }
+        },
+        CommitStatus::Committed(predecessor_record) => commit_record.compute_dependency(predecessor_record),
+        CommitStatus::Closed => CommitDependency::Independent,
+    };
+
+    match commit_dependency {
+        CommitDependency::Independent => (),
+        CommitDependency::DependentPuts { puts } => puts.into_iter().for_each(DependentPut::apply),
+        CommitDependency::Conflict(conflict) => return Err(IsolationError::Conflict(conflict)),
+    }
+
+    Ok(())
+}
+
 #[derive(Debug)]
-pub enum IsolationError {
+struct DependentPut {
+    flag: Arc<AtomicBool>,
```

Comment:
I'm confused by this API of 'flag' and 'value' both being booleans, please leave a comment here

Reply:
The `flag` is the put reinsert flag, and the value is what that flag needs to be set to. A pointer and a value.